src/js/burials/config/form.js

```diff
@@ -182,6 +183,20 @@ const formConfig = {
           uiSchema: {
             deathDate: currentOrPastDateUI('Date of death'),
             burialDate: currentOrPastDateUI('Date of burial (includes cremation or interment)'),
+            'view:burialDateWarning': {
+              'ui:description': burialDateWarning,
+              'ui:options': {
+                hideIf: formData => {
+                  // If they haven't entered a complete year, don't jump the gun and show the warning
+                  if (formData.burialDate && formData.burialDate.split('-')[0].length !== 4) {
```

Comment:
![image](https://user-images.githubusercontent.com/12970166/28089403-c3cd3cdc-663d-11e7-97f1-b00273bff07d.png)
Looks like regex is the way to go. Good call.